serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/servicerecord/ServiceRecordFactory.java

```diff
@@ -198,42 +226,77 @@ private String lookupPemCertificate(String orgnumber) {
 
-        switch (DpiMessageRouter.route(personResource, notification, forcePrint)) {
-            case DPI:
-                return Optional.of(new SikkerDigitalPostServiceRecord(false, properties, personResource, ServiceIdentifier.DPI,
-                        identifier, null, null));
-            case PRINT:
-                return createPrintServiceRecord(identifier, onBehalfOrgnr, token, personResource);
-            case DPV:
-            return createPostVirksomhetServiceRecord(identifier);
-            default:
-                return createPostVirksomhetServiceRecord(identifier);
+        Set<Process> digitalpostProcesses = processService.findAll(ProcessCategory.DIGITALPOST);
+        List<ServiceRecord> serviceRecords = Lists.newArrayList();
+        for (Process p : digitalpostProcesses) {
+            DpiMessageRouter.TargetRecord target;
+            if (p.getIdentifier().equals(properties.getDpi().getInfoProcess())) {
+                target = DpiMessageRouter.route(personResource, Notification.NOT_OBLIGATED);
+            } else if (p.getIdentifier().equals(properties.getDpi().getVedtakProcess())) {
+                target = DpiMessageRouter.route(personResource, Notification.OBLIGATED);
+            } else {
+                throw new ServiceRegistryException(String.format("Error processing unknown digitalpost process: %s", p.getIdentifier()));
+            }
+
+            switch (target) {
+                case DPI:
+                    serviceRecords.add(createDigitalServiceRecord(personResource, identifier, p));
+                    break;
+                case PRINT:
+                    serviceRecords.add(createPrintServiceRecord(identifier, onBehalfOrgnr, token, personResource, p));
+                    break;
+                case DPV:
+                default:
+                    serviceRecords.add(createPrintServiceRecord(identifier, onBehalfOrgnr, token, personResource, p));
+                    serviceRecords.add(createDigitalDpvServiceRecord(identifier, p));
+            }
         }
 
+        return serviceRecords;
+    }
+
+    private ServiceRecord createDigitalDpvServiceRecord(String identifier, Process process) {
+        ArkivmeldingServiceRecord dpvServiceRecord = ArkivmeldingServiceRecord.of(DPV, identifier, properties.getDpv().getEndpointURL().toString());
+        Process defaultArkivmeldingProcess = processService.getDefaultArkivmeldingProcess();
+        dpvServiceRecord.getService().setServiceCode(defaultArkivmeldingProcess.getServiceCode());
+        dpvServiceRecord.getService().setServiceEditionCode(defaultArkivmeldingProcess.getServiceEditionCode());
+        dpvServiceRecord.setProcess(process.getIdentifier());
+        DocumentType docType = documentTypeService.findByBusinessMessageType(BusinessMessageTypes.DIGITAL_DPV)
+                .orElseThrow(() -> new ServiceRegistryException(String.format("Missing DocumentType for business message type '%s'", BusinessMessageTypes.DIGITAL)));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "Missing DocumentType for business message type '%s'" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1192)